R/rows-dm.R

```diff
@@ -381,14 +389,22 @@ dm_rows_run <- function(x, y, rows_op_name, top_down, in_place, require_keys, pr
       autoinc_col = autoinc_col
     )
 
-    if (!is.null(autoinc_col)) {
+    if (!is.null(autoinc_col) && is_src_db(x)) {
       tbls <- align_autoinc_fks(tbls, x, table, res)
+    } else if (!is.null(autoinc_col) && !is_src_db(x)) {
+      op_results[[i]] <- res$x_new
+      # `if` is necessary, since in case y doesn't contain the AI column, the target
+      # table should still have the AI values, but no update of potentially existing
+      # FKs should take place
+      if (!is.null(get_autoinc_col(x, table, colnames(tbl)))) {
+        tbls <- align_autoinc_fks(tbls, x, table, res$ai_lookup)
+      }
     } else if (!in_place) {
       op_results[[i]] <- res
     }
   }
 
-  if (in_place) {
+  if (in_place && is_src_db(x)) {
```

Comment:
We should have `!in_place` always if `!is_src_db()` -- I don't understand the necessity of this and the above changes.